### PR TITLE
Added documentation about escaping values for $GITHUB_ENV file

### LINKED ISSUE #10788: How to properly escape values for $GITHUB_ENV file
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/actions/learn-github-actions/workflow-commands-for-github-actions#environment-files

### What part(s) of the article would you like to see updated?

I'm missing explanation how to properly escape values written to $GITHUB_ENV file. It would be also helpful to describe how this file is read.

### Additional information

I'm writing this issue based on my current experience.

I'm trying to set environment variable with value which starts with `$` and so far I haven't been able to do so.
Please see following workflow file as example: https://github.com/klinki/github-actions-for-packages/blob/main/.github/workflows/test-env-vars.yml

---
Edited by maintainer

### Content Plan

Here is the [writer's content plant ](https://github.com/github/docs/issues/10788#issuecomment-974559306) for this issue

### PR DESCRIPTION
**Pull request by riadhlaabidi**
### Why:

Closes #10788 

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Added documentation about escaping environment variables' values written to the `GITHUB_ENV` environment file.

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [ ] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Review by NostalgicGareth — COMMENTED** [review]
- [ ]

**Comment by cmwilson21** [comment]
@riadhlaabidi Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Review by jhosman — COMMENTED** [review]
@riadhlaabidi Thanks for opening this PR! I consulted the team, and we made some edits to add further context about single versus double quotes. 

I'll go ahead and apply those changes for you, and we'll get this merged! 🎉

File: content/actions/using-workflows/workflow-commands-for-github-actions.md [hunk 1]
```diff
@@ -584,6 +584,8 @@ console.log("The running PID from the main action is: " +  process.env.STATE_pro
 
 During the execution of a workflow, the runner generates temporary files that can be used to perform certain actions. The path to these files are exposed via environment variables. You will need to use UTF-8 encoding when writing to these files to ensure proper processing of the commands. Multiple commands can be written to the same file, separated by newlines.
 
+Most commands in the following examples use double quotes for echoing strings, which will attempt interpolate characters like `$` for shell variable names. To always use literal values in quoted strings, you can use single quotes instead.
```
Comment (jhosman) [inline content/actions/using-workflows/workflow-commands-for-github-actions.md:587]:
```suggestion
Most commands in the following examples use double quotes for echoing strings, which will attempt to interpolate characters like `$` for shell variable names. To always use literal values in quoted strings, you can use single quotes instead.
```

**Review by jhosman — APPROVED** [review]

**Pull request merged**